### ISSUE TITLE
Whenever I delete a German character (ä,ö,ü,ß) I get this error

### ISSUE THREAD

**Issue by tsaber** [jrnl-org/jrnl#1341]
Bug

### Environment:
```
jrnl: v2.8.1
Python: 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110]
OS: Linux 4.19.104-microsoft-standard
```
installed via pip3

### Current Behavior:
```
Traceback (most recent call last):
  File "/home/zava/.local/bin/jrnl", line 8, in <module>
    sys.exit(cli())
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/cli.py", line 35, in cli
    return run(args)
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/jrnl.py", line 71, in run
    write_mode(**kwargs)
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/jrnl.py", line 139, in write_mode
    raw = _write_in_editor(config)
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/jrnl.py", line 190, in _write_in_editor
    raw = get_text_from_stdin()
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/editor.py", line 56, in get_text_from_stdin
    raw = sys.stdin.read()
  File "/usr/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
```

### Expected Behavior:
```
[Entry added to default journal]
```

### Repro Steps

```
% jrnl --debug
DEBUG    root         Parsed args: Namespace(debug=True, preconfig_cmd=None, postconfig_cmd=None, filename=None, on_date=None, today_in_history=False, month=None, day=None, year=None, start_date=None, end_date=None, contains=None, strict=False, starred=False, limit=None, excluded=[], edit=False, delete=False, export=False, tags=False, short=False, config_override=[], text=[])
DEBUG    root         Reading configuration from file /home/zava/.config/jrnl/jrnl.yaml
DEBUG    root         Using configuration "{'colors': {'body': 'none', 'date': 'none', 'tags': 'none', 'title': 'none'}, 'default_hour': 9, 'default_minute': 0, 'editor': '', 'encrypt': True, 'highlight': True, 'indent_character': '|', 'journals': {'default': '/home/zava/.local/share/jrnl/journal.txt'}, 'linewrap': 79, 'tagsymbols': '@', 'template': False, 'timeformat': '%Y-%m-%d %H:%M', 'version': 'v2.4.5'}"
DEBUG    root         Using journal name: default
Password:
DEBUG    root         opened EncryptedJournal with 354 entries
DEBUG    root         Write mode: starting
[Writing Entry; on a blank line, press Ctrl+d to finish writing]

##ääööüqweqweDEBUG    root         Write mode: appending raw text to journal "default": ##ääööüüqweqwe
[Entry added to default journal]
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.9/logging/__init__.py", line 1079, in emit
    msg = self.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 923, in format
    return fmt.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 659, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.9/logging/__init__.py", line 363, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/zava/.local/bin/jrnl", line 8, in <module>
    sys.exit(cli())
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/cli.py", line 35, in cli
    return run(args)
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/jrnl.py", line 71, in run
    write_mode(**kwargs)
  File "/home/zava/.local/lib/python3.9/site-packages/jrnl/jrnl.py", line 151, in write_mode
    logging.debug("Write mode: completed journal.write()", args.journal_name, raw)
Message: 'Write mode: completed journal.write()'
Arguments: ('default', '##ääööüüqweqwe')
```
Write a couple of German letters, and delete one of them. If you don't delete one of them it works just fine.

**Comment by micahellison** [comment]
Hi there @tsaber, thanks for reporting this. I'm having a hard time reproducing this, though I do have a slightly different environment than you:
```
jrnl: v2.8.3
Python: 3.9.7 (default, Aug 30 2021, 19:19:13)
[GCC 5.4.0 20160609]
OS: Linux 4.4.0-19041-Microsoft
```
Would you mind also sharing which shell you're using? I think this part of jrnl might behave different depending on that.

Also, this question might sound strange, but: does it still work when you press backspace 2, 4, or 6 times, rather than 1, 3, or 5 times? These characters are read as UTF-8, where they are double-byte Unicode characters, and the only hypothesis I can come up with is that your shell is somehow truncating one of the two bytes when it evaluates the backspace against the string it sends.

**Comment by micahellison** [comment]
I'm closing this for now, but feel free to comment if you have more info. Anyone else with this problem is also encouraged to chime in.

**Comment by tsaber** [comment]
this is in zsh
I'm sorry I totally forgot about this.
Since then I updated jrnl and python and it somehow broke  encryption in my jrnl. 

```
jrnl
/home/zava/.local/lib/python2.7/site-packages/jrnl/EncryptedJournal.py:2: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.fernet import Fernet, InvalidToken
[Compose Entry; press Ctrl+D to finish writing]

ääPassword:
Traceback (most recent call last):
  File "/home/zava/.local/bin/jrnl", line 11, in <module>
    load_entry_point('jrnl==2.0.1', 'console_scripts', 'jrnl')()
  File "/home/zava/.local/lib/python2.7/site-packages/jrnl/cli.py", line 227, in run
    raw = " ".join(args.text).strip()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)
```

I think I can't reproduce until I have fixed this.

**Comment by micahellison** [comment]
HI @tsaber, no worries. From your traceback, it looks like the problem is that you're running the outdated jrnl version 2.0.1 (instead of latest 2.8.3) on Python 2.7 (the latest versions of jrnl all require Python 3.7+).

**Comment by tsaber** [comment]
@micahellison 
thanks for the reply.
I now have the latest python and the latest jrnl. I had it installed both via sudo pip and pip and that mucked it all up. I fixed it since then but it still happens. Your guess was correct:

I tested it with pressing backspace odd and even times and it does indeed make the difference.

![githubtest](https://user-images.githubusercontent.com/9378902/139400415-15be1914-80c2-4e0f-83fb-b1d071201c4e.jpg)
.

**Comment by tsaber** [comment]
same thing with bash
![image](https://user-images.githubusercontent.com/9378902/139495709-8191c75b-c12d-4f58-9f4a-d46c74159a43.png)

and with fish
![image](https://user-images.githubusercontent.com/9378902/139495898-83ba9ead-febe-4807-a8eb-6d87351cb9df.png)

**Comment by micahellison** [comment]
Hi @tsaber, thanks for all this info. I still can't figure out how to reproduce this on my end. I see from the top of this issue that you're probably using WSL - what terminal emulator are you using? e.g. cmd.exe, Windows Terminal, PowerShell, etc.?

Also, in your shell, could you run `locale` and paste the output here? That could affect the way Python interprets these characters.

**Comment by tsaber** [comment]
I use either ConEmu on Windows or the built-in WSL terminal emulator.

`LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=`
locale is en_US.UTF-8 that is correct.
Do you think changing it to de_DE.UTF-8 will stop this error from happening?
I would like to continue using the English locale but for bug-tracking's sake I would try.

`pi@raspberrypi:~ $ jrnl
Path to your journal file (leave blank for /home/pi/.local/share/jrnl/journal.txt):
Do you want to encrypt your journal? You can always change this later [y/N] y
Journal will be encrypted.
Enter password for new journal:
Enter password again:
Do you want to store the password in your keychain? [Y/n] y
Please set a password for your new keyring:
Please confirm the password:
Encrypted journal 'default' created at /home/pi/.local/share/jrnl/journal.txt
[Writing Entry; on a blank line, press Ctrl+d to finish writing]

allo #Traceback (most recent call last):
  File "/home/pi/.local/bin/jrnl", line 10, in <module>
    sys.exit(cli())
  File "/home/pi/.local/lib/python3.7/site-packages/jrnl/cli.py", line 35, in cli
    return run(args)
  File "/home/pi/.local/lib/python3.7/site-packages/jrnl/jrnl.py", line 71, in run
    write_mode(**kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/jrnl/jrnl.py", line 139, in write_mode
    raw = _write_in_editor(config)
  File "/home/pi/.local/lib/python3.7/site-packages/jrnl/jrnl.py", line 190, in _write_in_editor
    raw = get_text_from_stdin()
  File "/home/pi/.local/lib/python3.7/site-packages/jrnl/editor.py", line 56, in get_text_from_stdin
    raw = sys.stdin.read()
  File "/usr/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 6: unexpected end of data
pi@raspberrypi:~ $`

also happens on my raspberry pi.


And I'm sorry I keep forgetting about this in the meantime. I hope you are not bothered by this. I'll set a reminder to check up on this tmrw

**Comment by micahellison** [comment]
> Do you think changing it to de_DE.UTF-8 will stop this error from happening? I would like to continue using the English locale but for bug-tracking's sake I would try.

I think it could be worth trying, though I also understand wanting to keep it.

> also happens on my raspberry pi.

Is your pi also in the English locale?

In [the docs for sys.stdin](https://docs.python.org/3/library/sys.html#sys.stdin) (which is where this error is happening), it mentions a configuration value for processing input that is affected by the [PYTHONIOENCODING environment variable](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONIOENCODING). Maybe setting this to "UTF-8" or "cp1252" might affect the way it parses the input.

Could you try each of those and see if it affects this issue?

> And I'm sorry I keep forgetting about this in the meantime. I hope you are not bothered by this. I'll set a reminder to check up on this tmrw

Not at all! I appreciate the time you're putting into this and I understand that this work is all voluntary. Whatever you're willing to contribute helps.

I'm still not 100% sure it's related to code pages, but I'm also not sure what else to try. I still think that your terminal is somehow "chopping" double-byte UTF-8 characters in half, and I can even explain the theory why, but I don't know how to keep it from doing it.

Basically, the character it can't decode is 0xc3, as seen in your error message:
```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 6: unexpected end of data
```
And in this nice [unicode character table](https://www.utf8-chartable.de/unicode-utf8-table.pl?start=128&names=-&utf8=0x), we can see a lot of double-byte UTF-8 characters that start with 0xc3, including the ä, ö, and ü vowels in your test case. So you are definitely entering unicode into your terminal, but your backspace is being treated like it's not unicode.

**Comment by tsaber** [comment]
I will do as soon as I got time. My CS study is asking a lot right now :)
